contracts/libraries/FixedPoint96.sol

```diff
@@ -1,10 +1,103 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity ^0.8.19;
 
+import {FullMath} from "./FullMath.sol";
+import {SafeCast} from "./SafeCast.sol";
+
+// Using custom type UQ64x96 with Unsafe math on Q64.96 numbers
+type UQ64x96 is uint160;
+
+using {
+    add as +,
+    sub as -,
+    eq as ==,
+    neq as !=,
+    lt as <,
+    gt as >,
+    lte as <=,
+    gte as >=,
+    div as /,
+    mul as *
+} for UQ64x96 global;
+
+using FixedPoint96 for UQ64x96 global;
+using SafeCast for uint256;
+
+function add(UQ64x96 a, UQ64x96 b) pure returns (UQ64x96) {
+    return UQ64x96.wrap(UQ64x96.unwrap(a) + UQ64x96.unwrap(b));
+}
+
+function sub(UQ64x96 a, UQ64x96 b) pure returns (UQ64x96) {
+    return UQ64x96.wrap(UQ64x96.unwrap(a) - UQ64x96.unwrap(b));
+}
+
+function eq(UQ64x96 a, UQ64x96 b) pure returns (bool) {
+    return UQ64x96.unwrap(a) == UQ64x96.unwrap(b);
+}
+
+function neq(UQ64x96 a, UQ64x96 b) pure returns (bool) {
+    return UQ64x96.unwrap(a) != UQ64x96.unwrap(b);
+}
+
+function lt(UQ64x96 a, UQ64x96 b) pure returns (bool) {
+    return UQ64x96.unwrap(a) < UQ64x96.unwrap(b);
+}
+
+function gt(UQ64x96 a, UQ64x96 b) pure returns (bool) {
+    return UQ64x96.unwrap(a) > UQ64x96.unwrap(b);
+}
+
+function lte(UQ64x96 a, UQ64x96 b) pure returns (bool) {
+    return UQ64x96.unwrap(a) <= UQ64x96.unwrap(b);
+}
+
+function gte(UQ64x96 a, UQ64x96 b) pure returns (bool) {
+    return UQ64x96.unwrap(a) >= UQ64x96.unwrap(b);
+}
+
+function div(UQ64x96 a, UQ64x96 b) pure returns (UQ64x96) {
+    return UQ64x96.wrap((a.toUint256() * 2 ** 96 / b.toUint256()).toUint160());
+}
+
+function mul(UQ64x96 a, UQ64x96 b) pure returns (UQ64x96) {
+    return UQ64x96.wrap(FullMath.mulDiv(a.toUint256(), b.toUint256(), 2 ** 96).toUint160());
+}
+
 /// @title FixedPoint96
 /// @notice A library for handling binary fixed point numbers, see https://en.wikipedia.org/wiki/Q_(number_format)
 /// @dev Used in SqrtPriceMath.sol
 library FixedPoint96 {
     uint8 internal constant RESOLUTION = 96;
-    uint256 internal constant Q96 = 0x1000000000000000000000000;
+    uint256 internal constant ONE = 0x1000000000000000000000000;
+
```

Comment:
Hmm so instead of doing .wrap(0) you would prefer something like .wrap(FixedPoint96.ZERO) or you want a constant of type UQ96x64 called ZERO in which case it would look like FixedPoint96.ZERO.. I think both are slightly strange. The latter definitely because it's not really related at all to the FixedPoint lib

Reply:
yeah thats fair